front/app/api/custom_fields/useRawCustomFields.ts

```diff
@@ -0,0 +1,49 @@
+import fetcher from 'utils/cl-react-query/fetcher';
+import { useQuery } from '@tanstack/react-query';
+import customFieldsKeys from './keys';
+
+// typings
+import {
+  ICustomFields,
+  ICustomFieldInputType,
+  CustomFieldsKeys,
+} from './types';
+import { CLErrors } from 'typings';
+
+type Params = {
+  projectId?: string;
+  phaseId?: string;
+  inputTypes?: ICustomFieldInputType[];
+};
+
+export const fetchCustomFields = ({
+  projectId,
+  phaseId,
+  inputTypes,
+}: Params) => {
+  const apiEndpoint = phaseId
+    ? `admin/phases/${phaseId}/custom_fields`
+    : `admin/projects/${projectId}/custom_fields`;
+
+  return fetcher<ICustomFields>({
+    path: `/${apiEndpoint}`,
+    action: 'get',
+    queryParams: {
+      input_types: inputTypes,
+    },
+  });
+};
+
+const useRawCustomFields = ({ projectId, phaseId, inputTypes }: Params) => {
```

Comment:
Added this hook to request custom fields. Similar to the existing `useCustomFields` hook, but that one was doing all kinds of other stuff that I didn't need here. So I made a simplified version.